src/clj/puppetlabs/services/ca/certificate_authority_service.clj

```diff
@@ -27,9 +30,9 @@
              comidi/routes->handler)
          settings
          path
-         wrap-with-authorization-check
-         puppet-version)))
-   context)
+         (fn [request] (wrap-with-authorization-check request {:oid-map oid-mappings}))
```

Comment:
I might be wrong because I'm still pretty new to this codebase but I don't really having a problem with the service function Kevin originally suggested. tk-auth and the ca service seem related enough to not be overly weird in combining the oid mapping and the auth middleware function.

I agree that a whole new service isn't worth the headache and that the moistness is also not worth it, so the `auth-middleware-with-oid-mappings` type idea seems like a reasonable compromise.